### ISSUE TITLE
BUG: Fixtures with a connection name other than default will keep searching for the default connection host

### ISSUE THREAD

**Issue by genellern** [cakephp/cakephp#17715]
### Description

I have the following connections to the database.
```return [
    'Datasources' => [
        'default' => [
            'className' => Connection::class,
            'driver' => Mysql::class,
            'persistent' => false,
            'url' => env('MYSQL_URL'),
            'encoding' => 'utf8',
            'timezone' => 'UTC',
            'unix_socket' => null,
            'cacheMetadata' => true,
            'quoteIdentifiers' => true,
        ],
        'master' => [
            'className' => Connection::class,
            'driver' => Mysql::class,
            'persistent' => false,
            'url' => env('MYSQL_URL_MASTER'),
            'encoding' => 'utf8',
            'timezone' => 'UTC',
```

The test connections

```
        'test' => [
            'className' => Connection::class,
            'driver' => Mysql::class,
            'url' => env('MYSQL_URL_TEST'),
            'encoding' => 'utf8',
            'timezone' => 'UTC',
            'cacheMetadata' => true,
            'quoteIdentifiers' => true,
        ],
        'test_master' => [
            'className' => Connection::class,
            'driver' => Mysql::class,
            'persistent' => false,
            'url' => env('MYSQL_URL_TEST_MASTER'),
            'encoding' => 'utf8',
```

In the Fixture that uses the "master" connection, I set the connection name

```
public string $connection = 'test_master';
```

When I run the test, it will try to connect to host of the default connection. I stopped the default docker container with the database server. And I started getting the following error:

```
PDOException: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for itic_mysql failed: Name or service not known

```

itic_mysql is the name of the default connection host.

To work around this bug, I had to rename the fixture connection name in the `init()` method, like:

```
    public function init(): void
    {
        $this->connection = 'master';

        $this->records = [
            [
                'id' => Text::uuid(),
                'status' => 'created',
            ],
        ];
        parent::init();
    }
```

You can reach out to me if you have any other questions.

### CakePHP Version

5.0.1.

### PHP Version

8.2

**Comment by markstory** [comment]
> To work around this bug, I had to rename the fixture connection name in the init() method, like:

What is the value of `$this->connection` in `init()` before you redefined the connection name?

> And I started getting the following error:

Is there a stacktrace for this error?

**Comment by genellern** [comment]
The way the connections are setup locally is that I have two docker containers running mysql:8. I use the same environment variables, username, password and port for development and testing. The project uses 2 different databases (2 in dev, 2 in the tests db container) .  
One for the clients -the default connection- and one for the admin -the master database- site.

But I noticed the tests weren't passing for this particular case that uses the master connection.
 
I set the connection in the fixture like:
```
public string $connection = 'test_master';
```

My guess is that it's prefixing the connection name with the string `"test_"`

I'll try having a connection with the name `'test_test_master'`, comment out the line that renames the connection in the `init()` method and see if it connects successfully to the tests database server.


There is no stack trace other than it not being able to find the container hostname, the error I posted in the initial comment.

-  What is the value of $this->connection in init() before you redefined the connection name?
`´test_master´`

**Comment by markstory** [comment]
>  What is the value of $this->connection in init() before you redefined the connection name?    ´test_master´

Would you try upgrading to 5.0.9 temporarily and see if that resolves your issue? I saw that you're on 5.0.1 and the first couple 5.0 releases had connection aliasing flipped around by accident. See https://github.com/cakephp/cakephp/pull/17633

**Comment by genellern** [comment]
Apparently that solves the issue. I'll run some more tests but at first glance upgrading to the latest 5.x works fine.
